### DISCUSSION TITLE
Can you please make this project much more advance and simple by integrating with AI

### DISCUSSION THREAD

**Discussion by Greatz08** [getredash/redash#6788] [General]
What i mean by integrating with AI is that you know that sql is not something many knows and even if some knows they only know the basic stuff like crud operations and some more things but for complex things and to create complex chart we would require much more experience with sql and which max dont have and that is why they dont try these awesome projects, But right now many can use chatgpt and gemini and llama and other pretty great llms for writing advance sql queries with which we can generate charts and other things BUT still its not too much comfortable process and time consuming as many detailed info needs to be first provided to llm and then we have to explain it correctly for what we actually want and test that query with redash so not too great experience.
If redash can integrate with any foss vector database like weaviate,milvus,quadrant which can take sql databases too i guess(not 100% sure)and we can store and create embeddings for it and ask questions then it will easily understand everything of our database and then create accordingly and we wont have to feed every info to  it and it will be able to give proper response i believe.NOTE - this is just my idea i have to research more on it but just wanted to share it with you plus want you to work on it to make it more innovative and capable solution for max users and not just pro users :-)).
I would appreciate if i get reply back on this discussion because what i am asking for is not impossible because many foss projects have implemented ai thing and microsoft are integrating everything with openai and same will be case for google products soon so if we want to make this project more capable then this thing is neccessary to be implemented.

**Comment by justinclift** [comment]
> Can you please make this project much more advance and simple by integrating with AI?

You're welcome to submit a Pull Request with the changes needed for this.

**Comment by justinclift** [comment]
As a general thought, it might be useful to look into this after the hallucination problem with "AI" software is solved.  Until that time though, I personally feel it's just too risky.

Things like this are all too common: https://www.racket.news/p/i-wrote-what-googles-ai-powered-libel

Then there are the privacy and confidentiality issues of potentially sending customer data to external systems.  More risk there too. :frowning:
